libecole/src/reward/dualintegral.cpp

```diff
@@ -0,0 +1,51 @@
+#include "ecole/reward/dualintegral.hpp"
+#include "ecole/scip/model.hpp"
+#include "ecole/utility/chrono.hpp"
+
+namespace ecole::reward {
+
+namespace{
+  static auto dual_bound(scip::Model& model) {
```

Comment:
Thinking more about it, I don't really like having a function named `dual_bound` here, which sometimes return 0, which is not an actual dual bound. I would rather have the logic which checks for SCIP's stage directly inside the extract function.